src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -762,6 +771,29 @@ internal MethodSymbol TryFindDisposePatternMethod(BoundExpression expr, SyntaxNo
             return disposeMethod;
         }
 
+        /// <summary>
+        /// Binds an awaiter for asynchronous dispose
+        /// </summary>
+        /// <param name="node">The synatx node to bind for</param>
+        /// <param name="awaitKeyword">The await keyword of the syntax</param>
+        /// <param name="disposeMethodOpt">The dispose method to call, or null to use IAsyncDisposable.DisposeAsync</param>
+        /// <param name="diagnostics">Populated with any errors</param>
+        /// <param name="hasErrors">True if errors occured during binding</param>
+        /// <returns>An <see cref="AwaitableInfo"/> with the bound information</returns>
+        internal AwaitableInfo BindAsyncDisposeAwaiter(SyntaxNode node, SyntaxToken awaitKeyword, MethodSymbol disposeMethodOpt, DiagnosticBag diagnostics, ref bool hasErrors)
+        {
+            TypeSymbol taskType = disposeMethodOpt is null
+                                    ? this.Compilation.GetWellKnownType(WellKnownType.System_Threading_Tasks_ValueTask)
```

Comment:
>GetWellKnownType [](start = 55, length = 16)

Please add a test where ValueTask is made missing (see `MakeTypeMissing` test helper).